src/iceberg/util/string_util.h

```diff
@@ -46,4 +46,15 @@ class ICEBERG_EXPORT StringUtils {
   }
 };
 
+/// \brief Transparent hash function that supports std::string_view as lookup key
+///
+/// Enables std::unordered_map to directly accept std::string_view lookup keys
+/// without creating temporary std::string objects, using C++20's transparent lookup.
+struct ICEBERG_EXPORT string_hash {
+  using hash_type = std::hash<std::string_view>;
+  using is_transparent = void;
+
+  std::size_t operator()(std::string_view str) const { return hash_type{}(str); }
```

Comment:
```suggestion
struct ICEBERG_EXPORT StringHash {
  using hash_type = std::hash<std::string_view>;
  using is_transparent = void;
  
  std::size_t operator()(const char* str) const        { return hash_type{}(str); }
  std::size_t operator()(std::string_view str) const   { return hash_type{}(str); }
  std::size_t operator()(const std::string& str) const { return hash_type{}(str); }
};
```

Suggestions for naming consistency and completeness.